olp-cpp-sdk-dataservice-write/src/VersionedLayerClientImpl.cpp

```diff
@@ -390,57 +390,29 @@ VersionedLayerClientImpl::CancelBatch(const model::Publication& pub) {
 
 olp::client::CancellationToken VersionedLayerClientImpl::CancelBatch(
     const model::Publication& pub, CancelBatchCallback callback) {
-  std::string publicationId = pub.GetId().value_or("");
-  if (publicationId.empty()) {
-    callback(client::ApiError(client::ErrorCode::InvalidArgument,
-                              "Invalid publication", true));
-    return {};
-  }
-
-  auto self = shared_from_this();
-  auto cancel_context = std::make_shared<client::CancellationContext>();
-  auto id = tokenList_.GetNextId();
-  auto cancel_function = [=]() {
-    self->tokenList_.RemoveTask(id);
-    callback(client::ApiError(client::ErrorCode::Cancelled,
-                              "Operation cancelled.", true));
-  };
+  const auto publication = pub;
```

Comment:
This is a double copy. Is it necessary? You can just copy capture it in the lambda and it should be good.